library/profiler_builtins/build.rs

```diff
@@ -79,9 +79,12 @@ fn main() {
         cfg.define("COMPILER_RT_HAS_ATOMICS", Some("1"));
     }
 
-    // Note that this should exist if we're going to run (otherwise we just
-    // don't build profiler builtins at all).
-    let root = Path::new("../../src/llvm-project/compiler-rt");
+    // Get the LLVM `compiler-rt` directory from bootstrap.
+    println!("cargo:rerun-if-env-changed=RUST_COMPILER_RT_FOR_PROFILER");
+    let root = PathBuf::from(
+        env::var("RUST_COMPILER_RT_FOR_PROFILER")
```

Comment:
Could we default into the original path instead of erroring out here if the env. variable is missing? It's annoying that a normal build of the Rust crates in `compiler` or `library` does not work without setting many obscure required environment variables, this would add another one.

Reply:
I can have it fall back to `"../../src/llvm-project/compiler-rt"`, if you want, but bear in mind that doing so will still fail (or give weird results) if that directory doesn't contain a suitable copy of the LLVM source code.

Reply:
I know. It's a trade-off between "I need to checkout LLVM in order to build library" vs "I need to figure out that I need to use an environment variable, and then checkout LLVM and point the environment variable to the right directory". Since the first one is pre-existing behavior and avoids the need of yet another arbitrary ENV. variable needed to build compiler/library, I'd personally prefer it.

Reply:
Fair enough. And in the common failure case of the directory not existing at all, the other patch in this PR should mean that we at least end up with sensible error messages instead of bizarre linker errors caused by compiling zero source files.

Reply:
Indeed, that's a nice improvement :+1: